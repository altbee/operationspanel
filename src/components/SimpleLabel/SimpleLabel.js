import React from "react";
import PropTypes from "prop-types";

const SimpleLabel = props => {
  const { color, text, width, title } = props;

  return (
    <div
      title={title}
      style={{ width: `${width}px` }}
      className={`simple-label-component bg-${color || "blue"}`}
    >
      {text}
    </div>
  );
};

SimpleLabel.propTypes = {
  color: PropTypes.string.isRequired,
  text: PropTypes.oneOfType([PropTypes.string, PropTypes.object]).isRequired,
  title: PropTypes.string,
  width: PropTypes.number
};

SimpleLabel.defaultProps = {
  width: 100,
  title: ""
};

export default SimpleLabel;
