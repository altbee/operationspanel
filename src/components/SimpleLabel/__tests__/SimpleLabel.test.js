import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import SimpleLabel from "../SimpleLabel";

configure({ adapter: new Adapter() });

describe("<SimpleLabel />", () => {
  const colorList = ["red", "pink"];
  const textList = ["Hello World", "Hi there"];
  const defProps = {
    width: 100,
    title: ""
  };

  const props = {
    color: colorList[0],
    text: textList[0],
    width: 50,
    title: "title text"
  };

  it("renders when props are defined", () => {
    const rendered = shallow(<SimpleLabel {...props} />);

    expect(
      rendered.find(`div.simple-label-component.bg-${colorList[0]}`).exists()
    ).toBe(true);
    expect(rendered.find(`div[title='${props.title}']`).exists()).toBe(true);
    expect(rendered.find(`div.bg-${colorList[0]}`).text()).toEqual(textList[0]);
    expect(rendered.find(`div.bg-${colorList[0]}`).prop("style")).toEqual({
      width: "50px"
    });
  });

  it("renders default props when title and width are not supplied", () => {
    const rendered = shallow(
      <SimpleLabel color={colorList[1]} text={textList[1]} />
    );
    expect(
      rendered.find(`div.simple-label-component.bg-${colorList[1]}`).exists()
    ).toBe(true);
    expect(rendered.find(`div[title='${defProps.title}']`).exists()).toBe(true);
    expect(rendered.find(`div.bg-${colorList[1]}`).text()).toEqual(textList[1]);
    expect(rendered.find(`div.bg-${colorList[1]}`).prop("style")).toEqual({
      width: `${defProps.width}px`
    });
  });

  it("matches the snapshot", () => {
    const tree = shallow(<SimpleLabel {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
