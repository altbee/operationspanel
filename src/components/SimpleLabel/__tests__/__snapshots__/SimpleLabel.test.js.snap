// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SimpleLabel /> matches the snapshot 1`] = `
<div
  className="simple-label-component bg-red"
  style={
    Object {
      "width": "50px",
    }
  }
  title="title text"
>
  Hello World
</div>
`;
